File: Backend/WebApi/Models/SleepReview.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace WebApi.Models;

public partial class SleepReview
{
    public int Id { get; set; }

    public Guid UserId { get; set; }

    public int? WearableLogId { get; set; }

    public int? SurveyId { get; set; }

    public int? SleepSettingsId { get; set; }

    public DateTime CreatedAt { get; set; }

    public int? SmarterSleepScore { get; set; }

    public virtual Survey? Survey { get; set; }

    public virtual WearableData? WearableLog { get; set; }

    public virtual SleepSetting? SleepSetting { get; set; }
}

File: Frontend/Smarter_Sleep/smarter_sleep/lib/app/models/sleep_review.dart
import 'package:smarter_sleep/app/models/sleep_schedule.dart';

import 'survey.dart';
import 'wearable_log.dart';

class SleepReview {
  final int id;
  final String userId;
  final DateTime createdAt;
  final int smarterSleepScore;
  final Survey survey;
  final WearableLog wearableLog;
  final SleepSchedule? sleepSchedule;

  SleepReview({
    required this.id,
    required this.userId,
    required this.createdAt,
    required this.smarterSleepScore,
    required this.survey,
    required this.wearableLog,
    this.sleepSchedule,
  });

  factory SleepReview.fromJson(Map<String, dynamic> json) {
    return SleepReview(
        id: json['id'],
        userId: json['userId'],
        createdAt: DateTime.parse(json['createdAt']),
        smarterSleepScore: json['smarterSleepScore'],
        survey: Survey.fromJson(json['survey']),
        wearableLog: WearableLog.fromJson(json['wearableLog']),
        sleepSchedule: _parseSleepSchedule(json['sleepSetting']));
  }

  static SleepSchedule? _parseSleepSchedule(dynamic json) {
    if (json != null) {
      return SleepSchedule.fromJson(json);
    }
    return null;
  }
}
